api/extension-guides/language-model.md

```diff
@@ -19,35 +19,100 @@ The process for using the Language Model API consists of the following steps:
 
-To interact with a language model, extensions should first craft their prompt and then send a request to the language model. You can use prompts to provide instructions to the language model on the broad task that you're using the model for. Prompt can also define the context in which user messages are interpreted.
+To learn more about the concepts of prompt engineering, we suggest reading OpenAI's excellent [Prompt engineering guidelines](https://platform.openai.com/docs/guides/prompt-engineering).
 
-In the following example, the first message is used to specify the persona used by the model in its replies and what rules the model should follow.
-The second message then provides the specific request or instruction coming from the user. It determines the specific task to be accomplished.
+>**Tip:** take advantage of the rich VS Code extension API to get the most relevant context and include it in your prompt. For example, to include the contents of the active file in the editor.
+
+### Use the `LanguageModelChatMessage` class
+
+The Language Model API provides the `LanguageModelChatMessage` class to create user messages. A language model prompt consists of one or more user messages that define the context in which the user's message is interpreted.
+
+In the following example, the first message provides context for the prompt:
+
+- The persona used by the model in its replies (in this case, a cat)
+- The rules the model should follow when generating responses (in this case, explaining computer science concepts in a funny manner by using cat metaphors)
+
+The second message then provides the specific request or instruction coming from the user. It determines the specific task to be accomplished, given the context provided by the first message.
 
 ```typescript
 const craftedPrompt = [
-    new vscode.LanguageModelChatUserMessage('You are a cat! Think carefully and step by step like a cat would. Your job is to explain computer science concepts in the funny manner of a cat, using cat metaphors. Always start your response by stating what concept you are explaining. Always include code samples.'),
-    new vscode.LanguageModelChatUserMessage('I want to understand recursion')
+    new vscode.LanguageModelChatMessage.User('You are a cat! Think carefully and step by step like a cat would. Your job is to explain computer science concepts in the funny manner of a cat, using cat metaphors. Always start your response by stating what concept you are explaining. Always include code samples.'),
+    new vscode.LanguageModelChatMessage.User('I want to understand recursion')
 ];
 ```
 
-For prompt engineering, we suggest reading OpenAI's excellent [guidelines](https://platform.openai.com/docs/guides/prompt-engineering).
+### Use `@vscode/prompt-tsx`
```

Comment:
In my opinion it may be better to keep the readme of prompt-tsx as the point of truth for its usage, as duplicating knowledge here is likely to make lead to it being outdated. Or at most have a brief example (which is sort of what it currently is) and a link to its readme or npm page for complete information.

Reply:
Now that I think about it, I tend to agree.

So we could make this section leaner and high level (motivate usage of library) and make sure the link to the prompt-tsx readme prominent.

Reply:
I also set up a PR with more info into its readme https://github.com/microsoft/vscode-prompt-tsx/pull/50

Reply:
Removed this section as we have links to the README higher up in the article.